### ISSUE TITLE
support breaking up long LABEL lines e.g. with OCI labels

### ISSUE THREAD

**Issue by guillaume-d** [hadolint/hadolint#483]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

hadolint should parse the Dockerfile below completely without exiting early and exit with status 0.
`docker build` has no problem with it.
Apparently hadolint's parser seems able to recover after the first `=\` at the end of a line.
Note that the strange way to write LABEL seems to be the only syntactically valid way to break up long LABEL lines where even the label names are long, like OCI's (those mentioned in https://github.com/hadolint/hadolint/issues/480).

### Actual behavior

Instead it complains as follows:
```
Dockerfile.split-label:8:1 a non empty string 
```
### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.18.0-6-ga0d655d
```

**Dockerfile** (if relevant)

```Dockerfile
FROM scratch

LABEL org.opencontainers.image.authors="The Author" \
    org.opencontainers.image.description="Does as advertised."

LABEL org.opencontainers.image.authors="The Author" \
    org.opencontainers.image.description=\
"Does as advertised."

LABEL org.opencontainers.image.authors=\
"The Author" \
    org.opencontainers.image.description=\
"Does as advertised."

LABEL \
    org.opencontainers.image.authors=\
"The Author" \
    org.opencontainers.image.description=\
"Does as advertised."
```

**Additional environment details (OS, stack version, etc.)**

Linux

**Comment by judfs** [comment]
I also ran into "`./Dockerfile:166:1 a non empty string`" with my file that splits env across multiple lines:

```
# NOTE: The whitespace is significant.
ENV CLASSPATH=\
"/.../foo.jar"\
":/../bar.jar"\
    PYTHONPATH=\
"/.../woof"\
":/.../meow"\
    OTHER_ENV=1
```

```
hadolint --version
Haskell Dockerfile Linter 2.12.0
```
